localstack/services/dynamodb/models.py

```diff
@@ -12,6 +18,70 @@
 )
 
 
+@dataclasses.dataclass
+class TableStreamType:
+    """
+    When an item in the table is modified, StreamViewType determines what information is written to the stream for this table.
+    - KEYS_ONLY - Only the key attributes of the modified item are written to the stream.
+    - NEW_IMAGE - The entire item, as it appears after it was modified, is written to the stream.
+    - OLD_IMAGE - The entire item, as it appeared before it was modified, is written to the stream.
+    - NEW_AND_OLD_IMAGES - Both the new and the old item images of the item are written to the stream.
+    Special case:
+    is_kinesis: equivalent to NEW_AND_OLD_IMAGES, can be set at the same time as StreamViewType
+    """
+
+    stream_view_type: Optional[StreamViewType]
+    is_kinesis: bool
+
+    @property
+    def needs_old_image(self):
+        return self.is_kinesis or self.stream_view_type in (
+            StreamViewType.OLD_IMAGE,
+            StreamViewType.NEW_AND_OLD_IMAGES,
+        )
+
+    @property
+    def needs_new_image(self):
+        return self.is_kinesis or self.stream_view_type in (
+            StreamViewType.NEW_IMAGE,
+            StreamViewType.NEW_AND_OLD_IMAGES,
+        )
+
+
+class DynamoDbStreamRecord(TypedDict, total=False):
+    ApproximateCreationDateTime: int
+    SizeBytes: int
+    Keys: Key
+    StreamViewType: Optional[StreamViewType]
+    OldImage: Optional[AttributeMap]
+    NewImage: Optional[AttributeMap]
+    SequenceNumber: Optional[int]
+
+
+class StreamRecord(TypedDict, total=False):
+    """
+    Global record type, this can contain both a KinesisRecord and a DynamoDBStreams record?
+    """
+
+    eventName: str
+    eventID: str
+    eventVersion: str
+    dynamodb: DynamoDbStreamRecord
+    awsRegion: str
+    eventSource: str
+
+
+StreamRecords = list[StreamRecord]
+
+
+class TableRecords(TypedDict):
```

Comment:
nit: I'd add a short docstring-style commend to briefly explain what's a `TableRecords` and a `RecordMap`.